akka-actor-typed/src/main/scala/akka/actor/typed/internal/pubsub/TopicImpl.scala

```diff
@@ -147,14 +161,45 @@ private[akka] final class TopicImpl[T](topicName: String, context: ActorContext[
     case TopicInstancesUpdated(newTopics) =>
       context.log.debug("Topic list updated [{}]", newTopics)
       topicInstances = newTopics
+      if (lastActivityForTtl == Long.MinValue) {
+        ttlAndTimers.foreach {
+          case (ttl, timers, _) =>
+            timers.startTimerWithFixedDelay(TtlTick.asInstanceOf[Command[T]], ttl / 2L)
+        }
+      }
       this
 
     case GetTopicStats(replyTo) =>
       replyTo ! TopicStats(localSubscribers.size, topicInstances.size)
       this
 
+    case TtlTick =>
+      if (localSubscribers.isEmpty) {
+        // only ever arrives if ttl is defined
+        ttlAndTimers match {
+          case Some((ttl, _, clock)) =>
+            val limit = clock.currentTimeMillis() - ttl.toMillis
+            if (lastActivityForTtl < limit) {
+              context.log.debug(
+                "Topic [{}] reached TTL [{}] without activity, terminating",
+                topicName,
+                PrettyPrintableDuration(ttl))
```

Comment:
`PrettyPrintableDuration` doesn't have toString so this is not what you expect.  I think it's supposed to be used with the implicit and `pretty`.

Reply:
Ah, noticed in the registry PR but missed here...